python/ffsim/protocols/trace_protocol.py

```diff
@@ -45,3 +51,72 @@ def trace(obj: Any, norb: int, nelec: tuple[int, int]) -> float:
         "The object did not have a _trace_ method that returned the trace "
         "or a _diag_ method that returned its diagonal entries."
     )
+
+
+def _trace_term(
+    op: tuple[tuple[bool, bool, int], ...], norb: int, nelec: tuple[int, int]
+):
+    n_elec_alpha, n_elec_beta = nelec
+
+    combined_indices = set([(orb, spin) for _, spin, orb in op])
+    norb_alpha = sum([not spin for _, spin in combined_indices])
+    norb_beta = sum([spin for _, spin in combined_indices])
+
+    eta_alpha = 0
+    eta_beta = 0
+
+    # loop over the support of the operator
+    # assume that each site is either 0 or 1 at the beginning
+    # track the state of the site through the application of the operator
+    # if the state exceed 1 or goes below 0,
+    # the state is not physical and the trace must be 0
+    for i, spin in combined_indices:
+        initial_zero = 0
+        initial_one = 1
+        is_zero = True
+        is_one = True
+        for action, aspin, orb in reversed(op):
+            if (aspin, orb) != (spin, i):
```

Comment:
Let's choose a consistent naming scheme for the loop variables so that this comparison becomes more obvious. Something like
```python
if (this_spin, this_orb) != (spin, orb):
```
would be easier to read.